server/backend/api/app/routers/router_idim_proxy.py

```diff
@@ -0,0 +1,28 @@
+import logging
+from typing import List
+
+from api.app.integration.idim_proxy import IdimProxyService
+from api.app.requester import (Requester, get_current_requester,
+                               internal_only_action)
+from api.app.schemas import IdimProxySearchParamIdir
+from fastapi import APIRouter, Depends, Query
+
+ERROR_EXTERNAL_USER_ACTION_PROHIBITED = "external_user_action_prohibited"
+
+LOGGER = logging.getLogger(__name__)
+
+router = APIRouter()
+
+@router.get("/idir", response_model=dict, dependencies=[Depends(internal_only_action)])
+def idir_search(
+    user_id: str,
```

Comment:
probably clearer to call this something like 'search_user_id", to make it clear this is the user id we are searching for / looking up, instead of the current user.

Reply:
IDIM webservice is actually capable for searching not only userId, also username, lastname, firstname. However, our proxy currently only coded as userId only search to IDIM service.
For this router, the "user_id" in the argument actually will be translate to a "request parameter" not a "path parameter". So it will be "/search?user_id=ianliu" not "/search/user-id". It leaves more room to be expend in future if necessary without needing to refactor the name again.
But we can discuss.

Reply:
Discussed. Mark it resolved.